Makefile

```diff
@@ -71,7 +71,6 @@ apply: ## Deploy the controller into your ~/.kube/config cluster
 delete: ## Delete the controller from your ~/.kube/config cluster
 	helm template karpenter charts/karpenter --namespace karpenter \
 		$(HELM_OPTS) \
-		--set serviceAccount.create=false \
```

Comment:
Shouldn't the annotation be passed in with `--set` flag? Otherwise you're using the namespace default Service Account and giving it all the access. I guess for development it doesn't really matter, but this pattern was also copied everywhere as is often the case.

Reply:
We thought users would be using the getting started guide eksctl command to create the IAM Role for Service Accounts, which creates the service account and annotates with the role arn (https://karpenter.sh/v0.5.5/getting-started/#create-the-karpentercontroller-iam-role).   That might be a bad assumption for production installs, I'm not sure. But for dev installs, it's probably accurate and so leaving the `--set serviceAccount.controller.create=false` seems like the best dev workflow experience to me. Otherwise, we'd need some way of persisting the role arn in an env var or something and then using a set to annotate every time we `make apply`.

Reply:
That guide really needs re-writing to promote better practice, although from memory one of the other tabs has a better way of creating the roles and getting the ARN back? It's idiomatic for a Helm chart to create it's own service account(s), so I'd suggest documenting the externally created one, but with a name specified so it's not the default namespace one.

Reply:
For the dev workflow, once the role has been created it persists outside of the K8s resources. It'd be simplest to set it in the env.

Reply:
Yeah, looks like the terraform getting started guide does the annotation by passing to helm: https://karpenter.sh/v0.5.5/getting-started-with-terraform/#install-karpenter-helm-chart 

Speaking of which, it looks like that doc needs to be updated with this change since the `controller.clusterName` is now at the root level.

Reply:
also, agreed on the annotating. Can you update the docs and Makefile to orchestrate that process?

Reply:
@bwagner5 does the instance profile change make a difference to any of this? Secondly which docs need updating?

In a related side discussion I think the chart needs decoupling from the binary release so I'll open a new issue for this.

Reply:
FYI https://github.com/aws/karpenter/issues/1253

Reply:
The instance profile shouldn't make a difference since it's for the nodes that karpenter provisions and not the karpenter controller instance role.

Reply:
The docs that need updating are here (I think that's all of the places):

 - https://karpenter.sh/v0.6.0/getting-started/#install-karpenter-helm-chart (may want to move the helm install docs above the spot service linked role docs since we need to pass thru the annotation now. Probably want to add `--role-only` to the eksctl cmd to prevent creating the service account too)
 - https://karpenter.sh/v0.6.0/getting-started-with-terraform/#install-karpenter-helm-chart 
 - https://karpenter.sh/v0.6.0/development-guide/#environment-specific-setup

Reply:
@bwagner5 shouldn't we be waiting until there is a release to make the changes to the docs for that release? The `v0.6.0` chart has already been released so I don't think the docs shouldn't change for that version? This is one of the problems of directly linking your chart releases to your package releases.

While I wait for a response I'll update the non-docs references and undo the change to docs I accidentally made previously.

Reply:
@bwagner5 @ellistarn how do the docs get updated for a release?

Reply:
All docs before a release go in the preview directory https://github.com/aws/karpenter/tree/main/website/content/en/preview 

When we cut a release, the preview directory is copied to the version we are releasing, so it's fine to make the doc updates to preview in this PR.

Reply:
These should be updated now.